application/labels.go

```diff
@@ -1,16 +1,25 @@
 package application
 
-import "strings"
-
-// metaFromLabels creates app labels from zoidberg prefixed labels
-func metaFromLabels(labels map[string]string) map[string]string {
-	r := map[string]string{}
+import (
+	"log"
+	"strconv"
+	"strings"
+)
 
+// parseLabels creates app labels from zoidberg prefixed labels
+func parseLabels(labels map[string]string) map[int]map[string]string {
+	r := make(map[int]map[string]string)
 	for k, v := range labels {
-		if strings.HasPrefix(k, "zoidberg_meta_") {
-			r[strings.TrimPrefix(k, "zoidberg_meta_")] = v
+		if strings.HasPrefix(k, "zoidberg_port_") {
+			k2 := strings.TrimPrefix(k, "zoidberg_port_")
+			k3 := strings.SplitN(k2, "_", 2)
+			if p, err := strconv.Atoi(k3[0]); err != nil || len(k3) != 2 {
+				log.Printf("Found unparsable tag: %s", k)
```

Comment:
I did not use capital letters in logging, not sure what official position on this is.